test-suite/src/test/require-jdk8/org/scalajs/testsuite/javalib/time/LocalDateTest.scala

```diff
@@ -0,0 +1,790 @@
+      testTemporal(someDate.minusWeeks(-52))(of(2012, 2, 27))
+      testTemporal(someDate.minusWeeks(-53))(of(2012, 3, 5))
+      testTemporal(leapDate.minusWeeks(53))(of(2011, 2, 23))
+      testTemporal(leapDate.minusWeeks(52))(of(2011, 3, 2))
+      testTemporal(leapDate.minusWeeks(1))(of(2012, 2, 22))
+      testTemporal(leapDate.minusWeeks(-1))(of(2012, 3, 7))
+      testTemporal(leapDate.minusWeeks(-52))(of(2013, 2, 27))
+      testTemporal(leapDate.minusWeeks(-53))(of(2013, 3, 6))
+      testTemporal(MIN.minusWeeks(-104354999947L))(of(999999999, 12, 27))
+      testTemporal(MAX.minusWeeks(104354999947L))(of(-999999999, 1, 5))
+      expectThrows[DateTimeException](MIN.minusWeeks(1))
+      expectThrows[DateTimeException](MIN.minusWeeks(-104354999948L))
+      expectThrows[DateTimeException](MAX.minusWeeks(1043549999478L))
+      expectThrows[DateTimeException](MAX.minusWeeks(-1))
+      expectThrows[ArithmeticException](MIN.minusWeeks(Long.MaxValue))
+      expectThrows[ArithmeticException](MAX.minusWeeks(Long.MinValue))
+    }
+
+    it("should respond to `minusDays`") {
+      for (d <- samples)
+        testTemporal(d.minusDays(0))(d)
+      testTemporal(someDate.minusDays(365))(of(2010, 2, 28))
+      testTemporal(someDate.minusDays(1))(of(2011, 2, 27))
+      testTemporal(someDate.minusDays(-1))(of(2011, 3, 1))
+      testTemporal(someDate.minusDays(-365))(of(2012, 2, 28))
+      testTemporal(someDate.minusDays(-366))(leapDate)
+      testTemporal(leapDate.minusDays(366))(someDate)
+      testTemporal(leapDate.minusDays(365))(of(2011, 3, 1))
+      testTemporal(leapDate.minusDays(1))(of(2012, 2, 28))
+      testTemporal(leapDate.minusDays(-1))(of(2012, 3, 1))
+      testTemporal(leapDate.minusDays(-365))(of(2013, 2, 28))
+      testTemporal(leapDate.minusDays(-366))(of(2013, 3, 1))
+      testTemporal(MIN.minusDays(-730484999633L))(MAX)
+      testTemporal(MAX.minusDays(730484999633L))(MIN)
+      expectThrows[DateTimeException](MIN.minusDays(1))
+      expectThrows[DateTimeException](MIN.minusDays(-730484999634L))
+      expectThrows[DateTimeException](MAX.minusDays(730484999634L))
+      expectThrows[DateTimeException](MAX.minusDays(-1))
+      expectThrows[ArithmeticException](ofEpochDay(-2).minusDays(Long.MaxValue))
+      expectThrows[ArithmeticException](ofEpochDay(1).minusDays(Long.MinValue))
+    }
+
+    it("should respond to `adjustInto`") {
+      for {
+        d1 <- samples
+        d2 <- samples
+      } {
+        testTemporal(d1.adjustInto(d2))(d1)
+      }
+
+      for {
+        d <- samples
+        t <- Seq(LocalTime.MIN, LocalTime.MAX)
+      } {
+        expectThrows[DateTimeException](d.adjustInto(t))
+      }
+    }
+
+    it("should respond to `until`") {
+      val samples1 = samples ++ Seq(of(2012, 1, 29), of(2012, 1, 30), of(2012, 2, 28),
+          of(2013, 2, 28), of(2013, 3, 1), of(0, 12, 31), of(1, 1, 1))
+
+      for {
+        d <- samples1
+        u <- ChronoUnit.values() if u.isDateBased
```

Comment:
I created two `val`s in `TemporalTest` now: `dateBasedUnits` and `TimeBasedUnits`.